### PR TITLE
Cluster-Autoscaler - Kubernetes client config builder

### PR DESCRIPTION
**Pull request by mwielgus**

### CODE REVIEW

**Comment by fgrzadkowski** [comment]
Let's cc each other on all PRs. It'll be easier to track progress.

It probably makes sense to reference our umbrella bug (https://github.com/kubernetes/kubernetes/issues/24404) or other lower priority bugs.

@mwielgus Can you please create issues for tasks we've discussed?

@mwielgus @piosz

**Comment by piosz** [comment]
+1 to what @fgrzadkowski proposed

File: cluster-autoscaler/config/kubernetes.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+			return nil, err
+		}
+		kubeConfigOverride.ClusterInfo.InsecureSkipTLSVerify = insecure
+	}
+
+	return &kubeConfigOverride, nil
+}
+
+// GetKubeClientConfig returns rest client configuration based on the passed url.
+func GetKubeClientConfig(uri *url.URL) (*kube_rest.Config, error) {
```
Comment (piosz) [inline cluster-autoscaler/config/kubernetes.go]: add todo to clean this code

File: cluster-autoscaler/config/kubernetes.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	kube_rest "k8s.io/kubernetes/pkg/client/restclient"
+	kube_client_cmd "k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	kube_client_cmd_api "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+)
+
+const (
+	// APIVersion to be used.
+	APIVersion = "v1"
+
+	defaultKubeletPort        = 10255
```
Comment (piosz) [inline cluster-autoscaler/config/kubernetes.go]: not needed

File: cluster-autoscaler/config/kubernetes.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	defaultKubeletHttps       = false
```
Comment (piosz) [inline cluster-autoscaler/config/kubernetes.go]: same here

File: cluster-autoscaler/config/kubernetes.go (hunk 1 above)
Comment (mwielgus) [inline cluster-autoscaler/config/kubernetes.go]: Done.

File: cluster-autoscaler/config/kubernetes.go (hunk 2 above)
Comment (mwielgus) [inline cluster-autoscaler/config/kubernetes.go]: Done.

File: cluster-autoscaler/config/kubernetes.go (hunk 3 above)
Comment (mwielgus) [inline cluster-autoscaler/config/kubernetes.go]: Done.

**Comment by piosz** [comment]
LGTM

**Pull request merged**